core/server/common/src/main/java/alluxio/master/journal/checkpoint/CompoundCheckpointFormat.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal.checkpoint;
+
+import alluxio.master.journal.PatchedInputChunked;
+import alluxio.master.journal.checkpoint.CompoundCheckpointFormat.CompoundCheckpointReader.Entry;
+
+import com.esotericsoftware.kryo.io.InputChunked;
+import com.google.common.base.Preconditions;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Optional;
+
+/**
+ * Format for compound checkpoints. Compound checkpoints follow the format
+ *
+ * [componentName1, componentBytes1, componentName2, componentBytes2, ...]
+ *
+ * The bytes are written using Kryo's chunked encoding.
+ */
+public class CompoundCheckpointFormat implements CheckpointFormat {
+  @Override
+  public CompoundCheckpointReader createReader(CheckpointInputStream in) {
+    return new CompoundCheckpointReader(in);
+  }
+
+  @Override
+  public void parseToHumanReadable(CheckpointInputStream in, PrintStream out) throws IOException {
+    CompoundCheckpointReader reader = createReader(in);
+    Optional<Entry> entryOpt;
+    while ((entryOpt = reader.nextCheckpoint()).isPresent()) {
+      Entry entry = entryOpt.get();
+      out.printf("--- Begin checkpoint for %s ---", entry.getName());
+      CheckpointFormat format = entry.getStream().getType().getCheckpointFormat();
+      format.parseToHumanReadable(entry.getStream(), out);
```

Comment:
Should we handle the `TarballCheckpointFormat` here?

Reply:
it handles it by calling `TarballCheckpointFormat#parseToHumanReadable`, which tells the user to use the journal tool for more detail. There isn't a great way to represent a tarball in a `PrintStream`.